modernizer-annotation-processor/src/main/java/org/gaul/annotation_processsor/ModernizerAnnotationProcessor.java

```diff
@@ -0,0 +1,152 @@
+
+    @Override
+    public final SourceVersion getSupportedSourceVersion() {
+        return SourceVersion.latest();
+    }
+
+    @Override
+    public final boolean process(
+        Set<? extends TypeElement> annotations,
+        RoundEnvironment roundEnv
+    ) {
+        for (TypeElement annotation : annotations) {
+            List<String> annotatedClasses =
+                getAnnotatedClasses(roundEnv, annotation);
+            makeAnnotatedClassesFile(annotatedClasses);
+        }
+        return true;
+    }
+
+    private List<String> getAnnotatedClasses(
+        RoundEnvironment roundEnv,
+        TypeElement annotation
+    ) {
+        List<String> annotatedClasses = new ArrayList<String>();
+        for (Element element : roundEnv.getElementsAnnotatedWith(annotation)) {
+            List<String> warnings = Arrays.asList(
+                    element.getAnnotation(SuppressWarnings.class).value());
+            if (warnings.contains("modernizer")) {
+                if (element.getKind().isClass()) {
+                    annotatedClasses.add(getClassHeader(element));
+                }
+            }
+        }
+        return annotatedClasses;
+    }
+
+    private void makeAnnotatedClassesFile(List<String> annotatedClasses) {
+        if (annotatedClasses.isEmpty()) {
+            return;
+        }
+        FileWriter writer = null;
+        try {
+            File file = new File(
+                getOutputDirectory().getPath(),
+                ModernizerAnnotationOutput.IGNORE_CLASSES_FILE_NAME);
+            writer = new FileWriter(file);
+            for (String element : annotatedClasses) {
+                writer.write(element + "\n");
+            }
+            writer.close();
+        } catch (IOException e) {
+            throw new RuntimeException(e);
+        } finally {
+            if (writer != null) {
+                try {
+                    writer.close();
+                } catch (IOException e) {
+                    throw new RuntimeException(e);
+                }
+            }
+        }
+    }
+
+    private File getOutputDirectory() {
+        try {
+            FileObject fileObjectToGetPath =
+                processingEnv.getFiler().createResource(
+                    StandardLocation.CLASS_OUTPUT,
+                    "",
+                    "dummy-file.txt",
+                    null);
+            File outputDir =
+                new File(fileObjectToGetPath.getName()).getParentFile();
+            return ModernizerAnnotationOutput.getOutputPath(outputDir);
+        } catch (IOException e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    private String getClassHeader(Element classElement) {
+        String packageName = processingEnv
+            .getElementUtils()
+            .getPackageOf(classElement)
+            .getQualifiedName()
+            .toString()
+            .replace('.', '/');
+        final String classHeader = (!packageName.isEmpty() ?
```

Comment:
We generally put the operators (`?`, `:`) and this is pretty tough to read. If there's no good way to format this as is, maybe we need to throw the package stuff in a new variable, like `packagePrefix`